File: gnsals0904/level1/비밀지도.java
class Solution {
    static int size;
    public String[] solution(int n, int[] arr1, int[] arr2) {
        size = n;
        int[] resultArr = new int[n];
        String[] result = new String[n];
        for(int i = 0; i < n; i++){
            resultArr[i] = arr1[i] | arr2[i];
            result[i] = makeResult(resultArr[i]);
        }
        return result;
    }

    public String makeResult(int num){
        String result = Integer.toBinaryString(num);
        result = emptyReplaceZero(result);
        return numToHash(result);
    }

    public String emptyReplaceZero(String value){
        int sub = size - value.length();
        if(sub > 0){
            for(int i = 0; i < sub; i++){
                value = "0" + value;
            }
        }
        return value;
    }

    public String numToHash(String value){
        String result = "";
        for(int i = 0; i < size; i++){
            if(value.charAt(i) == '1'){
                result += "#";
            } else {
                result += " ";
            }
        }
        return result;
    }

}

File: yujin-level0/Week65/비밀지도.js
function solution(n, arr1, arr2) {
  const result = [];

  for (let i = 0; i < n; i++) {
    // 비트 OR 연산 후 2진수 변환 및 자리수 맞추기
    let binaryStr = (arr1[i] | arr2[i]).toString(2).padStart(n, "0");

    let row = "";
    for (let char of binaryStr) {
      row += char === "1" ? "#" : " ";
    }

    result.push(row);
  }

  return result;
}

// 다른 풀이
// function solution(n, arr1, arr2) {
//   return arr1.map((val, i) => {
//     // 두 지도의 숫자를 비트 OR 연산 후 2진수 문자열로 변환
//     // n자리 길이에 맞추어 앞쪽을 '0'으로 채움 (padStart)
//     // 1은 '#'으로, 0은 ' '으로 치환
//     return (val | arr2[i])
//       .toString(2)
//       .padStart(n, "0")
//       .replace(/1/g, "#")
//       .replace(/0/g, " ");
//   });
// }
